File: FFmpegPlayer/ViewController.m
//
//  ViewController.m
//  FFmpegPlayer
//
//  Created by youxiaobin on 2021/1/4.
//

#import "ViewController.h"
#import "FFPlayer.h"
#import "AdditionalView.h"

@interface ViewController()<AdditionalViewDelegate>
@property (nonatomic, strong)FFPlayer *player;
@end
@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _player = [[FFPlayer alloc] init];
    NSString *url = [[NSBundle mainBundle] pathForResource:@"test" ofType:@"ape"];
//    NSString *url = [[NSBundle mainBundle] pathForResource:@"audio" ofType:@"mp3"];
    [self.view addSubview:_player.renderView];
    _player.renderView.frame = CGRectMake(0, 0, 1280, 720);
    AdditionalView *additionalView = [[AdditionalView alloc] initWithFrame:_player.renderView.frame];
    additionalView.delegate = self;
    [self.view addSubview:additionalView];
    _player.ffPlayerDelegate = additionalView;
    [_player playWithUrl:url enableHWDecode:YES];
}

- (void)seekTo:(float)duration {
    [self.player seekTo:duration];
}
- (void)togglePlayAction {
    if(self.player.playState == FFPlayStatePlaying) {
        [self.player pause];
    } else {
        [self.player resume];
    }
}
- (void)pause {
    [self.player pause];
}
@end

File: FFmpegPlayer-Swift/ViewController.swift
//
//  ViewController.swift
//  FFmpegPlayer-Swift
//
//  Created by youxiaobin on 2021/1/4.
//

import Cocoa

class ViewController: NSViewController {
    
    private let player = FFPlayer.init()
    private let toolBar = ToolBarView.init()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if let url = Bundle.main.path(forResource: "1280x720", ofType: "mp4") {
            _ = self.player.play(url: url, enableHWDecode: true)
        }
        self.view.addSubview(self.player.displayRender)
        self.player.displayRender.frame = .init(x: 0, y: 0, width: 1280, height: 720)
        
        self.view.addSubview(toolBar)
        self.toolBar.frame = self.player.displayRender.frame;
        self.toolBar.delegate = self
        self.player.ffPlayerDelegate = self.toolBar
    }
}
extension ViewController: ToolBarViewProtocol {
    func seekTo(_ time: Float) {
        self.player.seekTo(time)
    }
    func togglePlayAction() {
        if self.player.playState() == .pause {
            self.player.resume()
        } else if self.player.playState() == .playing {
            self.player.pause()
        }
    }
    func pause() {
        self.player.pause()
    }
}

